validphys2/src/validphys/dataplots.py

```diff
@@ -921,6 +923,106 @@ def plot_replica_sum_rules(pdf, sum_rules, Q):
     return fig
 
 
+@figure
+def plot_polarized_momentum(pdf, Q, xmin=0.001, angular_momentum=False):
+    """
+    Plot the correlated uncertainties for the truncated integrals of the polarized
+    gluon and singlet distributions.
+    """
+    from matplotlib.patches import Ellipse
+
+    def compute_hists(preds, nbins=10):
+        binning = np.linspace(preds.min(), preds.max(), num=nbins)
+        frequencies, bins = np.histogram(preds, bins=binning, density=True)
+        return {"x": bins[:-1], "bins": bins, "weights": frequencies}
+
+    def compute_ellipse(preds_x, preds_y, nstd=3):
+        covmat = np.cov(preds_x, preds_y)
+        eigval, eigvec = np.linalg.eig(covmat)
+        sqrt_eigval = np.sqrt(eigval)
+
+        return {
+            "xy": (preds_x.mean(), preds_y.mean()),
+            "width": 2.0 * nstd * sqrt_eigval[0],
+            "height": 2.0 * nstd * sqrt_eigval[1],
+            "angle": np.degrees(np.arctan2(*eigvec[:, 0][::-1])),
+            "fill": False,
+            "linewidth": 2.0,
+            "color": "C0",
+        }
+
+    predictions = polarized_sum_rules(pdf, Q, lims=[(xmin, 1)])
+    if not angular_momentum:
+        xpreds = np.array(predictions["g"])
+        ypreds = np.array(predictions["singlet"]) / 2.0
+    else:
+        preds_low_x = polarized_sum_rules(pdf, Q, lims=[(1e-4, xmin)])
+        xpreds = np.array(predictions["g"]) + np.array(predictions["singlet"]) / 2.0
+        xpreds = 1 / 2 - xpreds  # substract the proton spin
+        ypreds = -(np.array(preds_low_x["g"]) + np.array(preds_low_x["singlet"]) / 2.0)
+
+    params = {
+        "width_ratios": [6, 1],
+        "height_ratios": [1, 6],
+        "nrows": 2,
+        "ncols": 2,
+        "sharex": "col",
+        "sharey": "row",
+    }  # Parameters that define the subplots
+    fig, ((ax_histx, ax_empty), (ax_scatr, ax_histy)) = plotutils.subplots(**params)
+    fig.subplots_adjust(wspace=1e-4, hspace=1e-1)
+
+    # Add the scatter plots
+    ax_scatr.scatter(xpreds, ypreds, label=pdf.label)
+    ax_scatr.scatter(xpreds.mean(), ypreds.mean(), marker="s", c="red")
+
+    # Add the Ellipse plot
+    ellipse = Ellipse(**compute_ellipse(xpreds, ypreds))
```

Comment:
If `compute_ellipse` is only used here, I think it would be better for the external function `_compute_ellipse` to return directly an `Ellipse` object.